Competition2017/src/competition/subsystems/BaseXCANTalonSpeedControlledSubsystem.java

```diff
@@ -0,0 +1,202 @@
+package competition.subsystems;
+
+import org.apache.log4j.Logger;
+
+import com.ctre.CANTalon.FeedbackDevice;
+import com.ctre.CANTalon.TalonControlMode;
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.command.PeriodicDataSource;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.math.PIDPropertyManager;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+/**
+ * We have a few Subsystems that have one primary goal:
+ * -Get a motor to rotate at a specific controlled speed. 
+ * -Occasionally set the motor to use a specific power.
+ * This requires a decent amount of overhead in terms of Properties and 
+ * other configuration, and is something that could be mostly delegated to this class,
+ * as the only significant point of difference is how many motors are controlled
+ * by the Subsystem.
+ * 
+ * If the system has more than one motor it is typically
+ * configured into follower mode, so almost no extra configuration is needed.
+ * @author John
+ *
+ */
+public abstract class BaseXCANTalonSpeedControlledSubsystem extends BaseSubsystem implements PeriodicDataSource {
+    
+    private static Logger log = Logger.getLogger(BaseXCANTalonSpeedControlledSubsystem.class);
+
+    protected final XCANTalon masterMotor;
+    protected XCANTalon followerMotor;
+       
+    // output telemetry properties
+    protected final DoubleProperty systemCurrentSpeed;
+    protected final DoubleProperty systemTargetSpeed;
+    protected final DoubleProperty systemOutputPower;
+    protected final DoubleProperty systemTalonError;
+    protected final BooleanProperty atSpeedProp;
+    protected final BooleanProperty enablesystemLogging;
+    
+    private final DoubleProperty systemSpeedThresh;
+    protected final PIDPropertyManager pidPropertyManager;
+    protected final double masterChannel;
```

Comment:
Why is this a double? It should only be converted immediately before being sent to the slave.